client/ayon_core/tools/loader/abstract.py

```diff
@@ -1,22 +1,71 @@
+"""Abstract base classes for loader tool."""
+from __future__ import annotations
+
 from abc import ABC, abstractmethod
-from typing import List
+from typing import List, Optional, TypedDict
```

Comment:
BTW `TypedDict` is available since python 3.8, is it ok?

Reply:
yes, you're right. Not sure about this. Older versions can be supported using `typing-extensions`.

Reply:
Or define the class under `typing.TYPECHECKING`